front/src/modules/simulationResult/components/ChartHelpers/enableInteractivity.tsx

```diff
@@ -140,10 +148,28 @@ const updateChart = <
       })
   );
 
+  // We need to redraw line with newY2() for SpaceCurveSlopes chart if Y2 exists
+  if (newY2) {
+    chart.drawZone.selectAll<SVGLineElement, T[]>('.additional-y').attr(
+      'd',
+      d3
+        .line<T>()
+        .x((d) => {
+          const key = supplementaryXAxis.find((axis) => d[axis as keyof T] !== undefined) || xAxis;
+          return newX(d[key as keyof T] as number | Date);
+        })
+        .y((d) => {
+          const key = supplementaryYAxis.find((axis) => d[axis as keyof T] !== undefined) || yAxis;
+          return newY2(d[key as keyof T] as number | Date);
+        })
```

Comment:
I think you can even simplify it like this:
```suggestion
        .x((d) => newX(d[xAxis as keyof T]))
        .y((d) => newY2(d['height' as keyof T]))
```